app/Models/Language/Language.php

```diff
@@ -462,100 +464,102 @@ public function scopeFilterableByNameOrAutonym($query, $name)
         });
     }
 
```

Comment:
@bradflood I agree with you and as part of the refactor, I have removed the logic from `AccessGroupFileset` and I have removed the methdos: `getQueryContentAvailable` and `scopeIsContentAvailableAndfilterableByMedia`. 

To improve the code, I have changed the Language Model to include the query in the `isContentAvailable` function rather than a separate method like `getQueryContentAvailable`. You can see this change in action at the following URL: https://github.com/faithcomesbyhearing/dbp/pull/904/files#diff-99a85e425daea5f8d6fc376bae4f0324fac349d7566732a794ed0b9c657097e9R662.

However, I have also added a new parameter named `bible_fileset_filters` to the `isContentAvailable` method in order to filter peer bible_filesets records. This addition is because, rather than creating a new join with the bible table to filter using bible_filesets, we can take advantage of the existing join with the bible table.

Using the isContentAvailable method with the bible_fileset_filters parameter is a good way to take advantage of the existing join with the bible table. However, I believe we could also use the scopeFilterableByMedia method.

Specifically, when DBP invokes the Language search endpoint and the query is an expensive one, we can use the `isContentAvailable` method with the `bible_fileset_filters` parameter. E.g. https://github.com/faithcomesbyhearing/dbp/pull/904/files#diff-99a85e425daea5f8d6fc376bae4f0324fac349d7566732a794ed0b9c657097e9R478

But, when the query is not expensive, such as when using the Language list, we can use the `isContentAvailable` method without the `bible_fileset_filters` parameter and instead use the `scopeFilterableByMedia` method. E.g.

https://github.com/faithcomesbyhearing/dbp/pull/904/files#diff-f2e57b521c1922993eb74d760d1f25995a8eccc0ae282dcf10578c611ac6c955R135